### PR TITLE
log SQL queries using Python's logging interface + add default logger

### PR DESCRIPTION
**Pull request by rutsky**
Now all unhandled Python log events are printed into `stderr`, instead of generating "No handlers could be found for logger" message.

### CODE REVIEW

**Comment by djmitche** [comment]
Looks good

**Comment by rutsky** [comment]
@djmitche I've added destroying connection pool on the thread. This should fix warnings (and should not break other tests :) ).

**Comment by djmitche** [comment]
Nice work!

**Comment by rutsky** [comment]
Ugh, tests are hanging now...

**Comment by rutsky** [comment]
@djmitche I decided to stick with simple solution: disable SQLite check for using DB from multiple threads.
Destroying pool from the thread doesn't work for some reason; for example in `buildbot.test.integration.test_upgrade.TestWeirdChanges.testUpgradeChangeNoRevision` test.

File: master/buildbot/db/enginestrategy.py [hunk 1]
```diff
@@ -89,6 +89,16 @@ def special_case_sqlite(self, u, kwargs):
             if not os.path.isabs(u.database[0]):
                 u.database = os.path.join(kwargs['basedir'], u.database)
 
+        else:
+            # For in-memory database SQLAlchemy will SingletonThreadPool and
```
Comment (djmitche) [inline master/buildbot/db/enginestrategy.py]: "..will use SingletonThreadPool.."

**Comment by djmitche** [comment]
There's a lot of black magic in the DB API!

LGTM.

**Comment by rutsky** [comment]
@djmitche can you take a look for the last time? :)

**Comment by rutsky** [comment]
@djmitche,  I decided to rework log catching enabling/disabling.
I'm trying to debug some stuff and noticed, that current method is not very convenient.

File: master/buildbot/db/enginestrategy.py [hunk 2]
```diff
@@ -89,6 +89,16 @@ def special_case_sqlite(self, u, kwargs):
             if not os.path.isabs(u.database[0]):
                 u.database = os.path.join(kwargs['basedir'], u.database)
```
Comment (sa2ajj) [inline master/buildbot/db/enginestrategy.py]: [nit] do you really need to keep this empty line here?

**Comment by sa2ajj** [comment]
LGTM

**Pull request merged**